src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientPackageTransformer.java

```diff
@@ -0,0 +1,140 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.csharp;
+
+import com.google.api.codegen.common.TargetLanguage;
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.config.InterfaceModel;
+import com.google.api.codegen.config.PackageMetadataConfig;
+import com.google.api.codegen.config.ProtoApiModel;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.transformer.FileHeaderTransformer;
+import com.google.api.codegen.transformer.GapicInterfaceContext;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.PackageMetadataNamer;
+import com.google.api.codegen.transformer.PackageMetadataTransformer;
+import com.google.api.codegen.transformer.StandardImportSectionTransformer;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.util.csharp.CSharpAliasMode;
+import com.google.api.codegen.viewmodel.PackageInfoView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.codegen.viewmodel.metadata.PackageMetadataView;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.base.Splitter;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CSharpGapicClientPackageTransformer implements ModelToViewTransformer<ProtoApiModel> {
+  private static final String CSPROJ_TEMPLATE_FILENAME = "csharp/gapic_csproj.snip";
+
+  private static final CSharpAliasMode ALIAS_MODE = CSharpAliasMode.Global;
+
+  private final GapicCodePathMapper pathMapper;
+  private final PackageMetadataConfig packageMetadataConfig;
+  private final CSharpCommonTransformer csharpCommonTransformer = new CSharpCommonTransformer();
+  private final FileHeaderTransformer fileHeaderTransformer =
+      new FileHeaderTransformer(new StandardImportSectionTransformer());
+  private final PackageMetadataTransformer metadataTransformer = new PackageMetadataTransformer();
+
+  public CSharpGapicClientPackageTransformer(
+      GapicCodePathMapper pathMapper, PackageMetadataConfig packageMetadataConfig) {
+    this.pathMapper = pathMapper;
+    this.packageMetadataConfig = packageMetadataConfig;
+  }
+
+  @Override
+  public List<ViewModel> transform(ProtoApiModel model, GapicProductConfig productConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new CSharpSurfaceNamer(productConfig.getPackageName(), ALIAS_MODE);
+    CSharpFeatureConfig featureConfig = new CSharpFeatureConfig();
+
+    InterfaceModel lastApiInterface = null;
+    for (InterfaceModel apiInterface : model.getInterfaces()) {
```

Comment:
You're exposed to an NPE if `getInterfaces` is empty. How about:
```java
List<ProtoInterfaceModel> apiInterfaces = model.getInterfaces();
if (apiInterfaces.isEmpty()) {
  return surfaceDocs;
}

GapicInterfaceContext context =
    GapicInterfaceContext.create(apiInterfaces.get(apiInterfaces.size() - 1), ...
```

Reply:
Will do.